### PR TITLE
feat(platform): getIssue

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Adds optional `getIssue()` function to `platform` with `useCache` exposed. Only needed for platforms which support interactive checkboxes.

## Context:

This will be used for upcoming Dashboard race condition fix.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
What about other platforms like `gitea` ?

**Comment by rarkins** [comment]
Does gitea have checkbox support now?

**Comment by viceice** [comment]
> Does gitea have checkbox support now?

Yes, will be official supported in v1.15.0, [PR](https://github.com/go-gitea/gitea/pull/15791) merged.

I've a custom javascript to support clickable checkboxes for older versions:

```js
/* eslint-env browser,es2020,greasemonkey */
/* global $,config */
(function () {
  'use strict';
  const prefix = '[VO] ';
  console.log(`${prefix}starting`);
  const cbRe = /^\s*- \[( |x)\] (.+)$/gm;

  try {
    const { owner, repo, type, id } = location.pathname?.match(/(?<owner>[^/]+)\/(?<repo>[^/]+)\/(?<type>pulls|issues)\/(?<id>\d+)/)?.groups ?? {};

    if (!owner) {
      console.log(`${prefix}issue / pr not found`);
      return;
    }

    if ($('#edit-title').length === 0) {
      console.log(`${prefix}no rights`);
      return;
    }

    /**
     *
     * @param {number} idx
     */
    async function process(idx) {
      /**
       * @type {{body:string}}
       */
      let { body } = await fetch(`/api/v1/repos/${owner}/${repo}/issues/${id}`).then((response) => response.json());
      const m = [...body.matchAll(cbRe)];

      if (!m[idx]) {
        console.log(`${prefix}not found: ${idx}`);
        return;
      }
      const line = m[idx];
      console.log(`${prefix}match:`, line[0]);
      body = body.replace(
        line[0],
        line[0].replace(/\[( |x)\]/, (_, val) => (val == ' ' ? '[x]' : '[ ]'))
      );

      body = JSON.stringify({ body });

      const headers = {
        'content-type': 'application/json',
        'X-Csrf-Token': config.csrf,
      };
      const resp = await fetch(`/api/v1/repos/${owner}/${repo}/issues/${id}`, {
        method: 'PATCH',
        headers,
        body,
      });

      if (!resp.ok) {
        return;
      }

      location.reload();
    }

    console.log(`${prefix}${owner}/${repo}/${type}#${id}`);

    $(document).on('click', '.timeline-item.comment.first .render-content .task-list-item > input', (e) => {
      if ($(e.target).closest('a').length > 0) return true;

      // disable checkboxes
      $('.timeline-item.comment.first .render-content .task-list-item > input').prop('disabled', true);

      const idx = $('.timeline-item.comment.first .render-content .task-list-item').index($(e.target).closest('.task-list-item'));
      console.log(`${prefix}click:`, idx);
      process(idx).catch((e) => console.error(`${prefix}unexpected error`, e));
    });

    // enable checkboxes
    $('.timeline-item.comment.first .render-content .task-list-item > input').prop('disabled', false);
  } catch (e) {
    console.error(`${prefix}unexpected error`, e);
  }
})();
```

😏

**Comment by rarkins** [comment]
Added gitea but code inspection only. @viceice please double check it

File: lib/platform/gitea/index.ts [hunk 1]
```diff
@@ -587,6 +587,16 @@ const platform: Platform = {
     return config.issueList;
   },
 
+  async getIssue(number: number, useCache = true): Promise<Issue> {
+    // istanbul ignore if
+    if (!useCache) {
+      delete config.issueList;
+    }
+    const issueList = await platform.getIssueList();
```
Comment (viceice) [inline lib/platform/gitea/index.ts:595]: This doesn't work as expected with `useCache=false` as `getIssueList` would use the cached http request. Sending a commit to fix it 🙃

**Review by viceice — APPROVED** [review]
Added same implementation to gitea like github

**Pull request merged**